NewForce-Cohort-9/tabloid-full-stack-full-stack-force
# Completed View All Tags

### Description
Added a feature to View All Tags from client that displays a list of tags for admin users (in the future) to be able to edit, delete, create from. For development purposes, added a Tag model, repository, and controller.

[Trello Ticket #13](https://trello.com/c/CvNNVb8q)

### Type of Change
-Added new, non-breaking feature.

### Testing Instructions


1. git add and git commit whatever you're working on on your branch
2. git checkout main
3. git fetch --a
4. git checkout feat-tags

If you don't see the most recent version of my files, git pull origin feat-tags.
If you haven't created the Tabloid database, please do so.

5. Use this to fill out tables and columns: [SQL data](https://github.com/NewForce-Cohort-9/tabloid-full-stack-full-stack-force/blob/main/SQL/01_Tabloid_Create_DB.sql)

6. Use this to seed the database: [SQL Seed Data](https://github.com/NewForce-Cohort-9/tabloid-full-stack-full-stack-force/blob/main/SQL/02_Tabloid_Seed_Data.sql)

7. Inside of Visual Studio (purple), click the execute with debugger solid green button to launch the app
If not logged in, you may login with any user in the database. I have been using [foo@bar.com](mailto:foo@bar.com) for the email parameter. Leave the password text box blank. Click "Login".

Click on “Tags” in the nav bar, this will redirect you to the /tags route. You should see a list of all tags sorted in alphabetical order. You should see an Edit button with an edit pencil icon, a Delete button with a Trashcan icon, and a button to Create New Tag above the list. None of these buttons are functional yet.  

Checklist:
[x] My code follows the style guidelines of this project
[x] I have performed a self-review of my code
[x] I have commented my code, particularly in hard-to-understand areas
[x] My changes generate no new warnings or errors
[x] I have added test instructions that prove my fix is effective or that my feature works
